tasks/util/invoke.py

```diff
@@ -115,6 +117,11 @@ def invoke_impl(user, func,
     else:
         headers = {}
 
+    # deletes warm set
+    if knative and reset:
+        r = redis.Redis(host='localhost', port=6379, db=0)
+        r.delete("w_{}/{}".format(user, func))
```

Comment:
This doesn't work in a distributed context, which is the primary mode of operation, hence probably shouldn't live here. There are separate tasks for managing Redis that work in all contexts, have a look at `redis.clear_queue`